crates/subspace-networking/src/create.rs

```diff
@@ -253,3 +421,60 @@ pub async fn create(
 
     Ok((node, node_runner))
 }
+
+// Builds the transport stack that LibP2P will communicate over along with an optional relay client.
+async fn build_transport(
+    keypair: identity::Keypair,
+    timeout: Duration,
+    yamux_config: YamuxConfig,
+    relay_transport: Option<ClientTransport>,
+) -> Result<Boxed<(PeerId, StreamMuxerBox)>, CreationError> {
```

Comment:
Nit: it would have been preferable to extract this function in a separate commit. Then it is much easier to follow what has actually changes.